### DISCUSSION TITLE
Parameterized dependencies

### DISCUSSION THREAD

**Discussion by victoraugustolls** [fastapi/fastapi#7776] [Questions]
### First check

* [X] I used the GitHub search to find a similar issue and didn't find it.
* [X] I searched the FastAPI documentation, with the integrated search.
* [X] I already searched in Google "How to X in FastAPI" and didn't find any information.

### Description

How can I create an async Parameterized dependency?

I found this [link](https://fastapi.tiangolo.com/advanced/advanced-dependencies/) but I'm not quite sure it is the best option to my use case.

I have an authentication dependency with the following signature:

`async def verify_token(request: Request):`

And I needed to pass a boolean as a parameter, like:

`async def verify_token(request: Request, get_details: boolean = False):`

So I though that the best way would be to create a class to do it, but I'm not quite sure about it.

Any thoughts about this? Thanks in advance!

**Comment by victoraugustolls** [comment]
I'm inclined to use a class dependency with an async call:

```
class Dependency:
	def __init__(self, get_details: bool = False):
        self._details = get_details

    async def __call__(self, request: Request):
		pass
```

Just wanted to make sure this is the best (and possibly only) option.

**Comment by tiangolo** [comment]
Yep @victoraugustolls that should work.

You could also create a function inside another, but I think this is probably cleaner.

**Comment by victoraugustolls** [comment]
I tried doing it this way so I don't have to instanciate 2 instances of my class but it didn't work. Any tips?

```
class Dependency:
	def __init__(self):
        self._details = get_details

    async def __call__(self, get_details: bool = False):
		async def decorator(request: Request):
			pass
		return decorator
```

**Comment by victoraugustolls** [comment]
Solved the issue on my end, thanks for the help!

**Comment by tiangolo** [comment]
Thanks for reporting back and closing the issue :+1:
